lib/spatial_features/unzip.rb

```diff
@@ -16,21 +16,19 @@ def self.paths(file_path, find: nil, **extract_options)
       return Array(paths)
     end
 
-    def self.extract(file_path, output_dir = Dir.mktmpdir, downcase: false)
+    def self.extract(file_path, tmpdir: nil, downcase: false)
+      tmpdir ||= Dir.mktmpdir
       [].tap do |paths|
         entries(file_path).each do |entry|
           next if entry.name =~ IGNORED_ENTRY_PATHS
           output_filename = entry.name
           output_filename = output_filename.downcase if downcase
-          path = "#{output_dir}/#{output_filename}"
+          path = "#{tmpdir}/#{output_filename}"
           FileUtils.mkdir_p(File.dirname(path))
           entry.extract(path)
           paths << path
         end
       end
-    rescue => e
```

Comment:
`remove_entry` is now called in `update_features!` since the tmp dir could be used for multiple zip files.